src/restc.erl

```diff
@@ -235,6 +235,17 @@ parse_type(Type) ->
         _ -> Type
     end.
 
+get_header(Header, Headers, Default) ->
+    NormalizedHeaders = lists:keymap(fun normalize_header/1, 1, Headers),
+    get_key(Header, NormalizedHeaders, Default).
+
+normalize_header(Header) ->
```

Comment:
Just always compare the result of string:lowercase/1 and you are done

Reply:
Maybe the comment is more appropriate for the function above but you get the idea.

Reply:
Aah, just saw that there's already a function in place that does just this. My bad for not digging deep enough!

Reply:
Have you noticed as well that there is a function called content-type/2 ?

Reply:
Good find, I wonder why the old code was using `get_key` which now seem quite pointless.

Reply:
So the content-type function is not exactly what you need.

But I like your previous approach with having a get_header function, I would implement that function like this:
```
get_header(Key, Headers, Default) ->
  SearchResult =
    lists:search(fun({OtherKey, _}) ->
                     string:lowercase(OtherKey) =:= string:lowercase(Key)
                 end, Headers),
  case SearchResult of
    {value, {_Key, Value}} -> Value;
    false -> Default
  end.
```

The default that you need to give is not the atom `json` but `<<"application/json">>`.

So the call would be what you had in the beginning:
```
parse_type(get_header(<<"Content-Type">>, Headers, <<"application/json">>)).
```

Reply:
Thanks for the suggestion. The current state of 18c69a7 seems to finally be bug free. I think that version is a bit easier to read and has less code. So if there is no performance concern with transforming the entire list beforehand I would prefer the current version.

Reply:
I believe I have another bug on my hands though.. I need to start writing tests for this thing.

Reply:
It's up to you. I'll approved it anyway, but the function get_header (without the default part) is a good candidate to be exported since it's useful for the users of restc. It's not only for internal use.